### PR TITLE
feat: Explore page with snapshot spaces

### PR DESCRIPTION
**Pull request by ChaituVR**
### Summary
Closes: #297 
- Explore page support for snapshot spaces
- Remove favorite button on all space cards

### How to test
- Update `.env` with `VITE_ENABLED_NETWORKS=s,eth,oeth,matic,arb1,sn`
- Go to http://localhost:8080/#/explore
- You should see new protocol selection menu and spaces related to the protocol

### CODE REVIEW

File: apps/ui/src/networks/offchain/index.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export function createOffchainNetwork(networkId: NetworkID): Network {
   return {
     readOnly: true,
     name: networkId === 's-tn' ? 'Snapshot (testnet)' : 'Snapshot',
-    avatar: '',
+    avatar: 'ipfs://bafkreifsxa2m5xzqx5jiq64oswbyyb6qmi6faykdjaipv2vsggkqf55gzm',
```
Comment (bonustrack) [inline apps/ui/src/networks/offchain/index.ts:66]: There is too less space around logo, but anyway we don't want to show Snapshot logo as a network for space preview

**Comment by bonustrack** [comment]
Can we show the badges too on space preview for Snapshot spaces?

**Comment by bonustrack** [comment]
Also can we use abbreviation for proposal and vote counters? With max 3 numbers like 652, 1.25m, 2.51k

File: apps/ui/src/views/My/Explore.vue [hunk 2]
```diff
@@ -1,20 +1,45 @@
...
 const spacesStore = useSpacesStore();
-
-onMounted(() => spacesStore.fetch());
-
 watchEffect(() => setTitle('Explore'));
+onMounted(() => spacesStore.fetch());
+const protocols = Object.values(explorePageProtocols).map((protocol: ProtocolConfig) => ({
+  key: protocol.key,
+  label: protocol.label
+}));
```
Comment (Sekhmet) [inline apps/ui/src/views/My/Explore.vue:14]: It feels a bit weird with all those put together without line breaks. Usually we separate different types of composables by whitespace.

Also I feel like protocols should be at the top, not bottom.

File: apps/ui/src/views/My/Explore.vue [hunk 3]
```diff
@@ -1,20 +1,45 @@
+import ProposalIconStatus from '@/components/ProposalIconStatus.vue';
```
Comment (Sekhmet) [inline apps/ui/src/views/My/Explore.vue:1]: This seems to be outside of <script> tag.

File: apps/ui/src/composables/useSpaces.ts [hunk 4]
```diff
@@ -15,6 +13,12 @@ export function useSpaces() {
   const loading = ref(false);
   const loadingMore = ref(false);
   const loaded = ref(false);
+  const protocol = ref('snapshot' as ExplorePageProtocol);
+
+  watch(protocol, toFilter => {
+    handleProtocolChange(toFilter);
+  });
+
```
Comment (Sekhmet) [inline apps/ui/src/composables/useSpaces.ts:21]: We tend to sort effects, so this would be at the bottom.

**Review by Sekhmet — COMMENTED** [review]
Looks like there are some linting issues in the code as well.

File: apps/ui/src/composables/useSpaces.ts (hunk 4 above)
Comment (ChaituVR) [reply apps/ui/src/composables/useSpaces.ts:21]: Interesting, but here it is in middle https://github.com/snapshot-labs/sx-monorepo/blob/master/apps/ui/src/composables/useMarkdownEditor.ts#L27 🙈 will change in `useSpaces` for now

File: apps/ui/src/stores/spaces.ts [hunk 5]
```diff
@@ -107,6 +111,8 @@ export const useSpacesStore = defineStore('spaces', () => {
     fetch,
     fetchMore,
     fetchSpace,
-    toggleSpaceStar
+    toggleSpaceStar,
+    protocol,
```
Comment (Sekhmet) [inline apps/ui/src/stores/spaces.ts:115]: nit. not sure if it makes sense, but we return and import those grouped by type (first refs, then functions), so protocol and handleProtocolChange shouldn't be next to each other.

File: apps/ui/src/composables/useSpaces.ts [hunk 6]
```diff
@@ -133,33 +134,43 @@ export function useSpaces() {
...
 
     await _fetchSpaces(false, filter);
 
     loadingMore.value = false;
   }
 
+  async function handleProtocolChange(newProtocol: ExplorePageProtocol) {
+    protocol.value = newProtocol;
+    await fetch();
+  }
```
Comment (Sekhmet) [inline apps/ui/src/composables/useSpaces.ts:156]: Not sure it makes sense to export two things that can be used to change protocol.

Now we can do it via `protocol` which will trigger this function, or via `handleProtocolChange`, which will trigger `protocol` ref change. I feel like just exposing `protocol` is enough as it seems confusing (and we don't use `handleProtocolChange` anyway, so we could just watch protocol and fetch on change.

Comment (ChaituVR) [reply apps/ui/src/composables/useSpaces.ts:156]: True that thank you, I was using this function in the `Explore` page before but failed to remove it later 😄

File: apps/ui/src/composables/useSpaces.ts [hunk 7]
```diff
@@ -66,7 +68,7 @@ export function useSpaces() {
         return network.api.loadSpaces(
           {
             skip: 0,
-            limit: SPACES_LIMIT
+            limit: explorePageProtocols['snapshotx'].limit
```
Comment (Sekhmet) [inline apps/ui/src/composables/useSpaces.ts:71]: Should it use hardcoded `snapshotx`?

File: apps/ui/src/composables/useSpaces.ts [hunk 8]
```diff
@@ -143,21 +144,28 @@ export function useSpaces() {
...
+    if (loading.value || !loaded.value || loadingMore.value) return;
     loadingMore.value = true;
 
     await _fetchSpaces(false, filter);
 
     loadingMore.value = false;
   }
 
+  watch(protocol, async (newProtocol: ExplorePageProtocol) => {
+    protocol.value = newProtocol;
```
Comment (Sekhmet) [inline apps/ui/src/composables/useSpaces.ts:156]: Do we need it? Looks like we are writing `protocol` second time when it's written to protocol before.

**Review by Sekhmet — COMMENTED** [review]
If I scroll all the way down it ends up in infinite loop.

https://github.com/snapshot-labs/sx-monorepo/assets/1968722/5ace7b84-0088-48e6-9d10-9644d54fb69d

Also, there will be only few `s` spaces, and then it ends up all as `s-tn`.

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**